### ISSUE TITLE
Better timeout indication on streaming json rpc requests

### ISSUE THREAD

**Issue by LukaszRozmej** [NethermindEth/nethermind#3629]
Some JSON RPC requests are streamed, lazy evaluated (eth_getLogs for example). If the response becomes too large it is send through multiple http responses in chunks. If the timeout occurs it just abruptly stops sending data.

It would be good to handle timeouts better (http codes?) for those scenarios. But I am not sure if its possible.

**Comment by LukaszRozmej** [comment]
@ice09 recommended solution is to split this filter into multiple by using some block numbers ranges.

For example in the merge when consensus client syncs deposit contract it uses block ranges of 1000.

**Comment by ice09** [comment]
Yes, thanks, that's what I did (for a on-site running node I used 10000) and it works very well. Will close this issue.

**Comment by LukaszRozmej** [comment]
we will still look into it

**Comment by LukaszRozmej** [comment]
we will buffer eth_getLogs for now

**Comment by kamilchodola** [comment]
Wasn't that already addressed? I recall we did something in that area.

@LukaszRozmej @rjnrohit

**Comment by rjnrohit** [comment]
Are you refeering to this issue: https://github.com/NethermindEth/nethermind/issues/7064 @kamilchodola  ?

**Comment by kamilchodola** [comment]
Yeah probably i recall this one @rjnrohit
